topcoffea/modules/utils.py

```diff
@@ -0,0 +1,17 @@
+import re
+
+# Match strings using one or more regular expressions
+def regex_match(lst,regex_lst):
+    # NOTE: For the regex_lst patterns, we use the raw string to generate the regular expression.
+    #       This means that any regex special characters in the regex_lst should be properly
+    #       escaped prior to calling this function.
+    # NOTE: The input list is assumed to be a list of str objects and nothing else!
+    matches = []
+    if len(regex_lst) == 0: return lst[:]
+    for s in lst:
+        for pat in regex_lst:
```

Comment:
This is actually an interesting point about using the python loops, since we are already extensively using numpy like features, although I don't know how I feel about forcing the function to take or return numpy arrays and I feel like converting from a python list to a numpy array and back would negate any potential speedups. At the end of the day I've always used this function with small-ish list sizes, but if there were some code that made many calls to this function that could still be reason for concern.

Reply:
You're right about having to use NumPy arrays. Since it's been running fine for you, I think we can just ignore this point.